.github/workflows/validateDevcontainer.yml

```diff
@@ -0,0 +1,140 @@
+name: Validate devcontainer
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  validate-devcontainer:
+    name: Validate devcontainer
+    runs-on: ubuntu-latest
+    
+    strategy:
+      matrix:
+        devcontainer-name: ["default"]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Make devcontainer '${{ matrix.devcontainer-name }}' singular
+        run: .devcontainer/devcontainer-helper --make-singular ${{ matrix.devcontainer-name }}
+
+      - name: Initialize devcontainer
+        uses: devcontainers/ci@v0.3
+        with:
+          push: never
+          runCmd: |
+            echo "Done."
+
+      # Even though devcontainers/ci should support $GITHUB_OUTPUT, it doesn't seem to work, so instead we write everything to a file that we later write to $GITHUB_OUTPUT in a non-devcontainer step.
+      - name: Get cache key parts
+        uses: devcontainers/ci@v0.3
+        with:
+          push: never
+          runCmd: |
+            git config --global --add safe.directory $PWD
+            rm -f .temporary-github-output
+            echo "llvm-hash=$(git rev-parse @:./llvm)" >> .temporary-github-output
+            . /etc/lsb-release
+            echo "ubuntu-codename=$DISTRIB_CODENAME" >> .temporary-github-output
+      - name: Export cache key parts
+        id: get-cache-key-parts
+        run: |
+          cat .temporary-github-output
+          cat .temporary-github-output >> $GITHUB_OUTPUT
+          rm .temporary-github-output
+        
+      - name: Restore ccache database
+        uses: actions/cache/restore@v3
+        with: 
```

Comment:
If you can back this with a suitable public hoisting for the Docker image, this might be a useful artifact for folks to use.

Is that something that could be done? Build a base image weekly-ish, push / host somewhere?

Anyway, it's unfortunate that this is so rough -- looks like that's just how it goes to make this work w/VSCode (?).

Helping VSCode users is a good goal, but I do not think is at all good to have that negatively impact others and am concerned about the maintenance burden (are you volunteering to be active code-owner?).

Having bash-y scripts in-tree to help folks having to re-invent this may be good, although I think most folks can fairly easily point VSCode at an appropriate C/C++ compiler, the build tree, and go from there? I don't think I needed all this when I last tried it, what's the benefit/strength of this?

Looks like this ccache wouldn't be useful for users anyway, only for purposes of automating checking that your development environment "works" -- at a glance doesn't seem to check the things that matters most re:checking that VSCode magic is sane -- otherwise I think this is mostly just build-checking in an ubuntu environment that's mostly like ones we already test?

Please consider handling that via external repository-- take a look at:

https://github.com/circt/update-circt
https://github.com/circt/images

and others.